### PR TITLE
feat(elixir): add ability to call and use rust filesystem vault

### PR DESCRIPTION
**Pull request by mikelodder7**

### CODE REVIEW

File: implementations/elixir/ockam/ockam_vault_software/test/vault_software_test.exs [hunk 1]
```diff
@@ -14,6 +14,13 @@ defmodule Ockam.Vault.Software.Tests do
     end
   end
```
Comment (nilslice) [inline implementations/elixir/ockam/ockam_vault_software/test/vault_software_test.exs:15]: @mikelodder7 -- do you mind adding a copy of this test and replacing the line `{:ok, handle} = SoftwareVault.default_init()` with `{:ok, handle} = SoftwareVault.file_init("/tmp")`

to see that it also produces the same hash? just to verify that the two vaults have consistent behavior.

Comment (mikelodder7) [reply implementations/elixir/ockam/ockam_vault_software/test/vault_software_test.exs:15]: sure

File: implementations/rust/vault/src/file/mod.rs [hunk 2]
```diff
@@ -138,6 +158,14 @@ impl DynVault for FilesystemVault {
 
     /// Export a secret key from the vault
     fn secret_export(&mut self, context: SecretKeyContext) -> Result<SecretKey, VaultFailError> {
+        use std::io::Write;
```
Comment (nilslice) [inline implementations/rust/vault/src/file/mod.rs:161]: just a little cleanup here from debugging :)

**Review by nilslice — CHANGES_REQUESTED** [review]
thx for fixing! let's remove the tmp file though and then will keep an eye out for the elixir lint fix

File: implementations/rust/vault/src/file/mod.rs (hunk 2 above)
Comment (mikelodder7) [reply implementations/rust/vault/src/file/mod.rs:161]: good catch

**Comment by nilslice** [comment]
@mikelodder7, these commit messages should also be updated to conform to the commit linter:

`feat(elixit): add ability to use filesystem vault` -> `feat(elixir): add ability to use filesystem vault` (misspell elixir)
`feat(rust): fix bug in filesystem vault` -> `fix(rust): fix bug in filesystem vault` (re-classify as fix)

**Review by nilslice — APPROVED** [review]
👏 👏 👏

**Pull request merged**